waspc/src/Wasp/Psl/Ast/Type.hs

```diff
@@ -0,0 +1,13 @@
+module Wasp.Psl.Ast.Type
+  ( Type (..),
+  )
+where
+
+import Wasp.Psl.Ast.Common (Name)
+import Wasp.Psl.Ast.Model (Body)
+
+data Type
+  = Type
+      Name
+      Body
```

Comment:
So does type actually support anything that can go into Model body, or is it a subset? That is relevant both here for AST and also for parser. If it is indeed anything that goes into model, then great. If not (and I would guess so, at least for view), then I would consider putting a comment here explaining that a bit.
Same goes for `view`.

Reply:
> So does type actually support anything that can go into Model body, or is it a subset? 

I'm only looking at this from the point of view of Prisma grammar and `type` and `view` are separate blocks. 

You can however, use composite types (`type` block) in models by using them as field types e.g.:
```prisma

model MyModel {
  id String @id
  someType SomeType
}

type SomeType {
  name String
  age Int
}
```

Reply:
Added some comments to both view and type.

Reply:
i meant in the sense that their ast/parser is subset of model's